src/librustc/ty/mod.rs

```diff
@@ -2322,6 +2322,20 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         }).borrow()
     }
 
+    /// Given the DefId of an item, returns its MIR, borrowed immutably.
+    /// Returns None if there is no MIR for the DefId
+    pub fn maybe_item_mir(self, did: DefId) -> Option<Ref<'gcx, Mir<'gcx>>> {
```

Comment:
Not all functions have a MIR available and `item_mir` panics if there isn't a MIR available for that def_id.